hail/python/test/hail/expr/test_expr.py

```diff
@@ -380,6 +380,10 @@ def test_aggregator_scope(self):
             t.aggregate(agg.filter(t.idx > 7, agg.sum(t.idx) / t.idx))
         with self.assertRaises(hl.expr.ExpressionException):
             t.aggregate(agg.group_by(t.idx % 3, agg.sum(t.idx) / t.idx))
+        with self.assertRaises(hl.expr.ExpressionException):
```

Comment:
I don't think this is actually testing the error that our users were making. There's no actual concrete aggregator here (one that generates a result rather than as a continuation function). Can you change your examples to do something like this?

```
hl.agg.counter(hl.agg.filter(hl.is_defined(mt.meta.pop), mt.meta.pop))
```

Note the `agg.counter` outside of the `agg.filter`.